### ISSUE TITLE
UI: Translation: Map selection: "Kartta", "Satelliitti" untranslated

### ISSUE THREAD

**Issue by ralfstrandell** [jariarkko/psgeo#112]
English translations of "Kartta" and "Satelliitti" are missing.

**Comment by ralfstrandell** [comment]
Note: "By default, the Maps JavaScript API uses the user's preferred language setting as specified in the browser"

If there is an explicit _?lang=XX_ query in the url however, then that language XX should override browser settings.

<script src="https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY&language=XX"></script>

**Comment by jariarkko** [comment]
Yeah this is an issue. Needs fixing. I need to figure out if there's a better way to set the language for google maps, because changing the URL would mean changing the current HTML. Or even calculating the URL dynamically.

**Comment by jariarkko** [comment]
This is tricky to fix in a general way, given limitations the API. I think I know a way to fix it, although it will have some unpleasant additional complex code in the HTML file. I will do this at some point (low priority) and test to see if we can make it well enough to be worth the effort.

**Comment by jariarkko** [comment]
One can only fire up Google Maps in specific language with the URL. But the URL isn't changeable in the "script src=" directive. One may have to do some dynamic loading, e.g., http://www.javascriptkit.com/javatutors/loadjavascriptcss.shtml

**Comment by ralfstrandell** [comment]
I wonder if this 'bug' is worth fixing at all.

Options:
1) fix it: https://gist.github.com/gbakernet/828536
2) trust the User Preferred Language set in the browser and hope that it matches with the language the user wants
3) hard-code language by using two html files: kartta.html and map.html, the only difference beeing the hard-coded language in the API call

#1 adds a lot of code to psgeo, #2 may cause problems too, #3 is simple.

**Comment by ralfstrandell** [comment]
In case 3, the psgeoDomainDefaults would need to be adjusted where paths are used to differentiate settings... but it would be a simple addition to a configuration line that already needs to be set by hand on every new site.

**Comment by ralfstrandell** [comment]
Fixed.

**Comment by jariarkko** [comment]
Thanks, this is indeed a solution!

(But maybe we should take it a bit slower. I think the code does the right thing, but there's also cases that it doesn't handle, and that's the reason why I had marked this as being worked on by me, as I wanted some time to think what's best. A full fix would be able to co-exist with other parameters and other languages. And preferably be shorter, because there are multiple instances of the HTML page, plus it would be nice for Psgeo example usage be short and sweet. A full solution may need to wait for refactoring of psgeo.js, because it would be nice to call on it to do the task. This isn't to say that you should jump and fix the above -- lets park this for a while and leave it open, because we have higher priority items to complete, and I'd like to return to this when the refactoring is done.)

**Comment by ralfstrandell** [comment]
Yes, of course, you are correct. This only fixes the issue for present two languages, and while it ẃould be simple to generalize, there would be too much clutter on the page among the html tags to look nice. Cleaner would be better. Sorry for not noticing that you had assigned this to yourself, I was too focused on the damn thing :/

**Comment by ralfstrandell** [comment]
FYI: related issue: solved.
The language code for Swedish has now been changed from incorrect se to correct sv. Even in this html file.

**Comment by jariarkko** [comment]
This is an improvement, thanks!

But the code doesn't work in all cases. I think we need we need to keep this one open,  and wait for the psgeo.js reorg so that we can call a function to do a full task with the parameters (and possibly also defaults, need to think about that). That way we can also make the html file compact, and do a one liner.

**Comment by ralfstrandell** [comment]
I stumbled upon this Stackoverflow post from Jan 2022:
https://stackoverflow.com/questions/901115/how-can-i-get-query-string-values-in-javascript

Now that our template html file already uses uiLanguage variable,
this might be an easy four line fix.
